honeybadgermpc/preprocessing.py

```diff
@@ -0,0 +1,157 @@
+import logging
```

Comment:
> there seems to be no universal set of preprocessing, instead it's likely we'll want all of the ones currently here and more
Agreed, I have structured the current code in a way that you can use a mix of them by generating and calling a get on the ones that you specifically need. The generation of course will be moved out later since that is expected to happen during the offline phase but the getters as in the current code should suffice.
> In general there are a few different ways to get preprocessing, one is for testing from pre-generated files, the other is as a result of output from a prior conversion step. Clearly for testing we want to simplify the procedure for using pre-generated files, but it's only one of the modes.
I agree, I completely forgot why we had a single source for generating the test preprocessed elements. I am going to fix that part so we reuse during tests.
> If we want to support multiple concurrent applications that use different preprocessing, they may each need to "reserve" a portion of the preprocessing data.
Haven't worked out on how this is going to happen - this would be one of the first steps for offline phase.
> Also, not all applications need all the preprocessing data. So maybe "wait for preprocessing" should be parameterized by how many of which kinds of preprocessing data we need?
With the current code, since we have to generate the preprocessed elements that we need, all we really need is one wait_for_processing when everything is ready, waiting for something specific doesn't seem to provide any benefit. In fact, with the actual offline phase implementation, this wait shouldn't be there since the preprocessed elements should be ready.